docs/docs/03-tools/10-daemon.md

```diff
@@ -0,0 +1,99 @@
+# Daemon Tools (Advanced)
+
+One advanced use case that GPTScript supports is daemon tools.
+A daemon tool is a tool that starts a long-running HTTP server in the background, that will continue running until GPTScript is done executing.
+Other tools can easily send HTTP POST requests to the daemon tool.
+
+## Example
+
+Here is an example of a daemon tool with a simple echo server written in an inline Node.js script:
+
+```
+Tools: my-daemon
+Param: first: the first parameter
+Param: second: the second parameter
+
+#!http://my-daemon.daemon.gptscript.local/myPath
+
+---
+Name: my-daemon
+
+#!sys.daemon node
+
+const http = require('http');
+
+const server = http.createServer((req, res) => {
+    if (req.method === 'GET' || req.method === 'POST') {
+        // Extract the path from the request URL
+        const path = req.url;
+
+        let body = '';
+
+        req.on('data', chunk => {
+          body += chunk.toString();
+        })
+
+        // Respond with the path and body
+        req.on('end', () => {
+          res.writeHead(200, { 'Content-Type': 'text/plain' });
+          res.write(`Body: ${body}\n`);
+          res.end(`Path: ${path}`);
+        })
+    } else {
+        res.writeHead(405, { 'Content-Type': 'text/plain' });
+        res.end('Method Not Allowed');
+    }
+});
+
+const PORT = process.env.PORT || 3000;
+server.listen(PORT, () => {
+    console.log(`Server is listening on port ${PORT}`);
+});
+```
+
+Let's talk about the daemon tool, called `my-daemon`, first.
+
+### The Daemon Tool
+
+The body of this tool begins with `#!sys.daemon`. This tells GPTScript to take the rest of the body as a command to be
+run in the background that will listen for HTTP requests. GPTScript will run this command (in this case, a Node script).
+GPTScript will assign a port number for the server and set the `PORT` environment variable to that number, so the
+server needs to check that variable and listen on the proper port. The server is required to respond to a GET request at
+`/` with 200 OK, so that GPTScript can make sure it is running properly before continuing execution.
```

Comment:
The path here is actually configurable. Do we want to mention that?

See the [UI](https://github.com/gptscript-ai/ui/blob/main/tool.gpt) for an example with the `(path ...)` option.

Reply:
Thanks for mentioning this - I wasn't aware this was a thing. I'll add it to the doc